lightning/src/ln/functional_unit_tests.rs

```diff
@@ -0,0 +1,969 @@
+// This file is Copyright its original authors, visible in version control
+// history.
+//
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE
+// or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// You may not use this file except in accordance with one or both of these
+// licenses.
+
+//! Trivial tests that mostly test some previously-incorrect behavior, but which rely on the
+//! functional test framework to simplify the test.
```

Comment:
I'm a bit meh on this module name. Feels wrong to include "functional" in the name just because they use the same setup utilities as `functional_tests.rs`. Also, it isn't really clear what should go in here vs other files. By "trivial tests" do you mean unit tests?  And if they are unit tests, should they go in `channelmanager.rs` or `channelmanager_tests.rs` file if that is getting too big? Assuming this is the unit being testing, otherwise they would belong somewhere else I'd imagine.

"previously-incorrect behavior" could have been a result of a gap in functional testing, too, so wouldn't necessarily belong here.

Reply:
> I'm a bit meh on this module name. Feels wrong to include "functional" in the name just because they use the same setup utilities as functional_tests.rs.

Fair, no strong opinion, we can drop that.

> Also, it isn't really clear what should go in here vs other files. By "trivial tests" do you mean unit tests?

Basically, yes.

> And if they are unit tests, should they go in channelmanager.rs or channelmanager_tests.rs file if that is getting too big?

Maybe, but many of them are specific to `channel` not `channelmanager`, or sometime a bit of both. Its not clear what the answer there would be.

> "previously-incorrect behavior" could have been a result of a gap in functional testing, too, so wouldn't necessarily belong here.

Yea, fair, in this case its just cause its testing exactly and only one specific line of code.

Reply:
> Maybe, but many of them are specific to `channel` not `channelmanager`, or sometime a bit of both. Its not clear what the answer there would be.
> 

Gotcha. I'm mostly concerned with having the tests grouped in some logically consistent way. And I feel the setup portion of the test shouldn't dictate how they are grouped. This would include preparing any data to feed into what is tested.

Not sure if that helps answer anything. What tests are you seeing as specific to `channel` or both?

Reply:
> And I feel the setup portion of the test shouldn't dictate how they are grouped. 

Yea, I guess I buy that, but TBH I'm not sure how *else* to group things. Like, we have a few rough groups but none of it really makes much sense, is already pretty arbitrary, and the bulk of our tests don't really fall into any of our categories anyway :(. I'm okay with dropping the changes here, but we do need to keep making *some* kind of forward progress in keeping `functional_tests` from growing unbounded.

> What tests are you seeing as specific to channel or both?

I think most of the tests here are really `channel` unit tests, eg the new test here is testing for an underflow in a function in `channel`, though hitting it via the `ChannelManager` API.

Reply:
Would you be ok with renaming this to `channel_tests.rs` and moving any tests from `channel.rs` here?

Reply:
In principle, yes, I'd want to go look at which tests, if any, were testing more stuff from channelmanager. Will do that.

Reply:
Hmm, no, it looks like a good chunk of the tests here are testing minor checks that are in `channelmanager`, not `channel`, so I don't think `channel_tests` captures it either.

Reply:
Just dropped the whole commit for now.